lib/format/formatter/get-data.js

```diff
@@ -0,0 +1,115 @@
+const parseArgs = require('../parse-args')
+const get = require('lodash.get')
+
+const getData = (input = {}, ...rest) => {
+  const args = Object.assign({scope: 'input'}, parseArgs(rest[1]))
+  const scopedData = input[args.scope] || {}
+  return get(scopedData, args.identifier)
+}
+
+const getArgs = (identifier, scope = 'input', input = {}) => {
+  const scopedData = input[scope] || {}
+  return get(scopedData, identifier)
+}
+
+const massageValueArgs = (identifier, args, updatedArgs) => {
+  identifier = identifier.trim()
+  let formatArgs = {}
+  let scopeParts = identifier.split('@')
+  if (scopeParts[1]) {
+    formatArgs.scope = scopeParts[0]
+    identifier = scopeParts[1]
+  }
+  if (identifier.includes('.')) {
+    formatArgs.nested = true
+  }
+  if (identifier.includes('[')) {
+    formatArgs.nested = true
+  }
+  if (!formatArgs.scope && !formatArgs.nested) {
+    return
+  }
+
+  const formatValue = getArgs(identifier, formatArgs.scope, args)
+
+  let formatString = '__FORMATDATA'
+  if (formatArgs.scope) {
+    formatString += `__SCOPE_${formatArgs.scope}`
+  }
+
+  const adjustedArgString = identifier
+    .replace(/\./g, '__dot__')
```

Comment:
`bar@foo.bar[0]` . 
=> . 
`__FORMATDATA__SCOPE_bar__IDENTIFIER_foo__dot__bar__leftbracket__0__rightbracket__`

As messageformat only allows keys that exist on the top-level of the context passed, we have to flatten the context and create normalised keys to access it. It could be anything really, but there remains some semblance of legibility normalising so

Reply:
Thank you 👍️